packages/messaging-client/src/Messaging.js

```diff
@@ -914,7 +918,7 @@ class Messaging {
 
         return conv2LastMessage.msg.created - conv1LastMessage.msg.created
       })
-      .slice(Number(offset) || 0, Number(limit) || 10)
+      .slice(startIndex, endIndex)
```

Comment:
nice catch, seems that previous slice had an error where end index didn't account for non 0 start index